libs/core-ui/src/lib/Cohort/CohortBasedComponent.ts

```diff
@@ -0,0 +1,77 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+import React from "react";
+
+import { IDataset } from "../Interfaces/IDataset";
+import { IFilter, ICompositeFilter } from "../Interfaces/IFilter";
+import { JointDataset } from "../util/JointDataset";
+
+import { Cohort, ErrorDetectorCohortSource } from "./Cohort";
+import { CohortStats } from "./CohortStats";
+import { ErrorCohort } from "./ErrorCohort";
+
+export interface ICohortBasedComponentProps {
+  dataset: IDataset;
+}
+
+export interface ICohortBasedComponentState {
+  baseCohort: ErrorCohort;
+  cohorts: ErrorCohort[];
+  selectedCohort: ErrorCohort;
+  jointDataset: JointDataset;
+}
+
+export abstract class CohortBasedComponent<
+  TProps extends ICohortBasedComponentProps,
+  TState extends ICohortBasedComponentState
+> extends React.PureComponent<TProps, TState> {
+  public updateSelectedCohort(
```

Comment:
any reason this was moved here?

Reply:
Yes! I noticed that there are lots of cohort-related operations underlying our dashboards (model-assessment and error-analysis, but also interpret). There's a whole bunch of things that they could just have in common. For now, this is only from model-assessment, because the `IDataset` is only used by model-assessment (because I haven't migrated others to use it yet). My goal was to put everything that's this general into core-ui so that everyone can build on it with the same foundation. That will require a bit more work in the future, of course, but this is a first step. What do you think?